Tasks/UsePythonV1/usepythonversion.ts

```diff
@@ -0,0 +1,146 @@
+
+import { Platform } from './taskutil';
+import * as toolUtil  from './toolutil';
+import { desugarDevVersion, pythonVersionToSemantic } from './version';
+
+interface TaskParameters {
+    version: string | undefined,
+    architecture: string | undefined
+}
+
+// Python has "scripts" or "bin" directories where command-line tools that come with packages are installed.
+// This is where pip is, along with anything that pip installs.
+// There is a seperate directory for `pip install --user`.
+//
+// For reference, these directories are as follows:
+//   macOS / Linux:
+//      <sys.prefix>/bin (by default /usr/local/bin, but not on hosted agents -- see the `else`)
+//      (--user) ~/.local/bin
+//   Windows:
+//      <Python installation dir>\Scripts
+//      (--user) %APPDATA%\Python\PythonXY\Scripts
+// See https://docs.python.org/3/library/sysconfig.html
+
+function binDir(installDir: string, platform: Platform): string {
+    if (platform === Platform.Windows) {
+        return path.join(installDir, 'Scripts');
+    } else {
+        return path.join(installDir, 'bin');
+    }
+}
+
+function pypyNotFoundError(majorVersion: 2 | 3) {
+    throw new Error([
+        task.loc('PyPyNotFound', majorVersion),
+        // 'Python' is intentional here
+        task.loc('ToolNotFoundMicrosoftHosted', 'Python', 'https://aka.ms/hosted-agent-software'),
+        task.loc('ToolNotFoundSelfHosted', 'Python', 'https://go.microsoft.com/fwlink/?linkid=871498')
+    ].join(os.EOL));
+}
+
+// Note on the tool cache layout for PyPy:
+// PyPy has its own versioning scheme that doesn't follow the Python versioning scheme.
+// A particular version of PyPy may contain one or more versions of the Python interpreter.
+// For example, PyPy 7.0 contains Python 2.7, 3.5, and 3.6-alpha.
+// We only care about the Python version, so we don't use the PyPy version for the tool cache.
+
+function usePyPy(majorVersion: 2 | 3, parameters: TaskParameters, platform: Platform): void {
+    const findPyPy = tool.findLocalTool.bind(undefined, 'PyPy', majorVersion.toString());
+    let installDir: string | null = findPyPy(parameters.architecture);
+
+    if (!installDir && platform === Platform.Windows) {
+        // PyPy only precompiles binaries for x86, but the architecture parameter defaults to x64.
+        // On Hosted VS2017, we only install an x86 version.
+        // Fall back to x86.
+        installDir = findPyPy('x86');
+    }
+
+    if (!installDir) {
+        // PyPy not installed in $(Agent.ToolsDirectory)
+        throw pypyNotFoundError(majorVersion);
+    }
+
+    // For PyPy, Windows uses 'bin', not 'Scripts'.
+    const _binDir = path.join(installDir, 'bin');
+
+    // On Linux and macOS, the Python interpreter is in 'bin'.
+    // On Windows, it is in the installation root.
+    const pythonLocation = platform === Platform.Windows ? installDir : _binDir;
+    task.setVariable('pythonLocation', pythonLocation);
```

Comment:
I'd like to remove this behavior for V1.  The idea was that it would give people a way to find the installation if they don't want to add it to PATH.  But we are getting rid of that option as well, and I don't think it's really that useful in practice.